### PR TITLE
Updating darc dependencies and enabling closure

### LINKED ISSUE #44078: Update Assembly Versions to 6.0.0.0
- [x] Update assemblyVersions to 6.0.0.0
- [x] Package index needs to be updated for assemblies which are added in the timeframe between the 6.0 snap and this change.
- [ ] Reenable the validation after a round trip change

cc @ericstj @ViktorHofer @safern

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/44078

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]

**Review by safern — APPROVED** [review]

File: src/installer/tests/Assets/TestUtils/TestProjects.targets [hunk 1]
```diff
@@ -9,19 +9,19 @@
 
   <ItemGroup>
     <KnownFrameworkReference Include="Microsoft.NETCore.App"
-      DefaultRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      DefaultRuntimeFrameworkVersion="6.0.0-alpha.1.20602.4"
       IsTrimmable="true"
-      LatestRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      LatestRuntimeFrameworkVersion="56.0.0-alpha.1.20602.4"
```
Comment (ViktorHofer) [inline src/installer/tests/Assets/TestUtils/TestProjects.targets:14]: 56

**Comment by ericstj** [comment]
Looks like libraries build is failing with
```
C:\Program Files\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(324,5): error NETSDK1127: (NETCORE_ENGINEERING_TELEMETRY=Build) The targeting pack Microsoft.NETCore.App is not installed. Please restore and try again.
```
I think this has to do with https://github.com/dotnet/runtime/blob/7e9dee84a91560dc8e3fda8bf1ddef664d07aa99/Directory.Build.targets#L31

However I wouldn't expect us to be using the *built-in* ref pack for `net6.0`.  We should hook that and use the one that was just built.

**Comment by ViktorHofer** [comment]
// Auto-generated message

https://github.com/dotnet/runtime/commit/69e114c1abf91241a0eeecf1ecceab4711b8aa62 which was merged 12/7 removed the intermediate src/coreclr/**src**/ folder. This PR needs to be updated as it touches files in that directory which causes conflicts.

To update your commits you can use this bash script: https://gist.github.com/ViktorHofer/6d24f62abdcddb518b4966ead5ef3783. Feel free to use the comment section of the gist to improve the script for others.

**Comment by safern** [comment]
> I think this has to do with

People are hitting this error whenever they don't have the rc1 targeting pack installed.
https://github.com/dotnet/runtime/issues/45755

**Pull request closed without merging**